libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -639,7 +639,14 @@ def write_I2C_config(f):
             % (n, n, n, n))
     f.write('#define HAL_I2C_DEVICE_LIST %s\n\n' % ','.join(devlist))
 
-
+def parse_timer(str):
+    '''parse timer channel string, i.e TIM8_CH2N'''
+    result = re.match(r'TIM([0-9]{1,2})_CH([1234])(N?)', str)
```

Comment:
I think that might not be quite right. It seems to allow for TIM0 (which can't exist) and not allow for TIM14 (which does exist)
Maybe just parse it as a number then check if the number is valid separately?

Reply:
Thank you for review.
It will allow for TIM14. Actually any one or two digit number is allowed.
Now I have added basic sanity check as you suggested. More precise check is done in pin definition table anyway.